File: 백태균/week06/132267_콜라_문제.java
class Solution {
    public int solution(int a, int b, int n) {
        int answer = 0;

        while (n >= a) {  // a 미만이면 콜라를 받을 수 없으니깐
            int div = (n / a) * b;  // 현재 n개를 a만큼 먹고 받을 수 있는 콜라의 수
            int remain = n % a;  // n개를 a만큼 먹고 남은 콜라의 수

            answer += div;
            n = div + remain;  // n을 받은 콜라의 수와 남은 콜라의 수로 갱신
        }

        return answer;
    }
}

File: 김진효/week06/132267_콜라_문제.py
# [성능 요약] 메모리: 11.3 MB 시간: 0.29 ms 

# 입력: 요구 빈 병 수(a), 증정 콜라 수(b), 가지고 있는 빈 병 수(n)
def solution(a, b, n):
    answer = 0
    while n >= a:
        plus = n // a
        n = plus * b + n % a
        answer += plus * b
    return answer

print(solution(2,1,20))